packages/blade/src/components/Input/PhoneNumberInput/PhoneNumberInput.web.tsx

```diff
@@ -0,0 +1,216 @@
+  const mergedRef = useMergeRefs(ref, inputRef);
+
+  const inputWrapperRef = React.useRef<HTMLDivElement | null>(null);
+  const [shouldShowClearButton, setShouldShowClearButton] = React.useState(false);
+  const [selectedCountry, setSelectedCountry] = React.useState<CountryCodeType>(defaultCountryCode);
+
+  React.useEffect(() => {
+    setShouldShowClearButton(Boolean(defaultValue ?? value));
+  }, [defaultValue, value]);
+
+  const renderTrailingInteractionElement = (): React.ReactNode => {
+    if (!shouldShowClearButton) return null;
+    return (
+      <IconButton
+        size="medium"
+        icon={CloseIcon}
+        onClick={() => {
+          if (isEmpty(value) && inputRef.current) {
+            // when the input field is uncontrolled take the ref and clear the input and then call the onClearButtonClick function
+            if (inputRef.current instanceof HTMLInputElement) {
+              inputRef.current.value = '';
+              inputRef.current.focus();
+            }
+          }
+          // if the input field is controlled just call the click handler and the value change shall be left upto the consumer
+          onClearButtonClick?.();
+          inputRef?.current?.focus();
+          setShouldShowClearButton(false);
+        }}
+        isDisabled={isDisabled}
+        accessibilityLabel="Clear Input Content"
+      />
+    );
+  };
+
+  const flags = React.useMemo(() => getFlagsForAllCountries(), []);
+  const countryData = React.useMemo(() => {
+    if (countries) {
+      return countries.map((countryCode) => {
+        return {
+          code: countryCode,
+          name: countryNameFormatter.of(countryCode)!,
+        };
+      });
+    }
+
+    return (Object.keys(flags) as CountryCodeType[])
+      .map((countryCode) => {
+        return {
+          code: countryCode,
+          name: countryNameFormatter.of(countryCode)!,
+        };
+      })
+      .sort((a, b) => a.name.localeCompare(b.name));
+  }, [countries, flags]);
+
+  const handleOnChange = ({
+    name,
+    value,
+    selectedCountry,
+  }: {
+    name?: string;
+    value?: string;
+    selectedCountry: CountryCodeType;
+  }): void => {
+    onChange?.({
+      name: name!,
+      value: value!,
+      phoneNumber: value ? formatPhoneNumber(value, selectedCountry) : undefined!,
+      dialCode: getDialCodeByCountryCode(selectedCountry),
+      countryCode: selectedCountry,
+    });
+  };
+
+  const onItemClick = ({ name }: { name: string }): void => {
+    const selectedCountry = name as CountryCodeType;
+    setSelectedCountry(selectedCountry);
+    handleOnChange({
+      selectedCountry,
+      name: inputRef.current?.name,
+      value: inputRef.current?.value,
+    });
+    inputRef.current?.focus();
+  };
+
+  return (
+    <BaseInput
+      setInputWrapperRef={(node) => {
+        inputWrapperRef.current = node as HTMLInputElement;
+      }}
+      ref={mergedRef}
+      id="textinput"
```

Comment:
nit: make it phone-number-input? so we can differentiate with textinput

Reply:
Weird, why do we even need this? even on TextInput.

Reply:
Done.